beacon_node/lighthouse_network/src/service/mod.rs

```diff
@@ -1401,7 +1393,10 @@ impl<E: EthSpec> Network<E> {
         let peer_id = event.peer_id;
 
         // Do not permit Inbound events from peers that are being disconnected, or RPC requests.
-        if !self.peer_manager().is_connected(&peer_id) {
+        if !self.peer_manager().is_connected(&peer_id)
```

Comment:
How about adding a comment here based on Pawan’s feedback, to clarify why we need the extra conditions?

https://github.com/sigp/lighthouse/pull/6400#discussion_r1764200888
> the reason we had the extra condition iirc is that we wanted to generate a RpcFailed event for HandlerErr::Outbound to send to sync and allow sync to remove the peer from all mappings